### PR TITLE
Add a better error message for TogetherAI 400 errors

### PR DESCRIPTION
**Pull request by wpietri**
Under certain weird cases, TogetherAI will return a 400 error with important details in the HTTP body. This makes it throw something that includes the content. Not sure if there's a better exception to throw than Exception here; I rummaged a bit and the only thing I found was the together response_status_exception function, which just throws Exception.

### CODE REVIEW

File: plugins/together/tests/test_client_behavior.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from unittest import mock
```
[the remark this answers is not in the document]
Comment (wpietri) [reply plugins/together/tests/test_client_behavior.py:1]: Sure thing!

File: plugins/together/newhelm/suts/together_client.py [hunk 2]
```diff
@@ -36,7 +36,12 @@ def _retrying_post(url, headers, json_payload):
     )
     session.mount("https://", HTTPAdapter(max_retries=retries))
     response = session.post(url, headers=headers, json=json_payload)
-    response_status_exception(response)
+    if response.status_code == 400:
+        raise Exception(
+            f"Unexpected TogetherAI 400 failure at {url} for {json_payload} with body {response.text}"
+        )
+    else:
+        response_status_exception(response)
```
[the remark this answers is not in the document]
Comment (wpietri) [reply plugins/together/newhelm/suts/together_client.py:44]: Sure! I knew I was ignoring a bigger picture here. The reason I didn't go for something fancier in this PR was that doing the broader case would require me to probe this API (or a few APIs) for enough examples to build a proper test suite. (E.g., When do they have response text? Do they put things in the response reason? Are the headers relevant?) Happy to do that, but maybe in a separate PR?

[the remark this answers is not in the document]
Comment (brianwgoldman) [reply plugins/together/newhelm/suts/together_client.py:44]: Of the two suggestions, I think the first is the most relevant to consider for this PR. Specifically if we think that we can generically handle errors across all SUTs, we don't need something special here in Together.

> doing the broader case would require me to probe this API (or a few APIs) for enough examples to build a proper test suite

In the first recommendation, I'm not sure that is needed. The open question is if the `sut_request` provides enough context that you could have solved the problem. If not, then lets go with this PR as is.

**Comment by wpietri** [comment]
The specific problem I'm trying to solve is that our service providers seem to be pretty quirky. Issues with adding SUTs are hard to diagnose because some of the feedback from services gets swallowed, and the retries can make take a long time to diagnose. The way I normally deal with quirky services  is to build up a set of tests that document real-world issues, so that anybody working on the code gets instant feedback.

I'm not at all attached to the solution here, just the test. So I'd say feel free to do anything you like with the code as long as the test case is preserved.

**Pull request merged**